relayer/processor/path_processor_internal.go

```diff
@@ -685,8 +686,30 @@ func (pp *PathProcessor) assembleAndSendMessages(
 	src, dst *pathEndRuntime,
 	messages pathEndMessages,
 ) error {
+	var needsClientUpdate bool
 	if len(messages.packetMessages) == 0 && len(messages.connectionMessages) == 0 && len(messages.channelMessages) == 0 {
-		return nil
+		var consensusHeightTime time.Time
+		if dst.clientState.ConsensusTime.IsZero() {
+			h, err := src.chainProvider.QueryIBCHeader(ctx, int64(dst.clientState.ConsensusHeight.RevisionHeight))
+			if err != nil {
+				return fmt.Errorf("failed to get header height: %w", err)
+			}
+			consensusHeightTime = time.Unix(0, int64(h.ConsensusState().GetTimestamp()))
+		} else {
+			consensusHeightTime = dst.clientState.ConsensusTime
+		}
+		if (dst.clientState.TrustingPeriod.Milliseconds() - time.Since(consensusHeightTime).Milliseconds()) < pp.clientUpdateThresholdTime.Milliseconds() {
```

Comment:
Implementation of my suggestion in https://github.com/cosmos/relayer/pull/1008/files#r1000932480

```suggestion
		if (float64(dst.clientState.TrustingPeriod.Milliseconds())*2/3 < float64(time.Since(consensusHeightTime).Milliseconds())) ||
			(time.Since(consensusHeightTime).Milliseconds() < pp.clientUpdateThresholdTime.Milliseconds()) {
```